test/shacheck.c

```diff
@@ -57,14 +59,24 @@ int main (int argc, char *argv[]) {
     char *cmd = untaint(argv[1]);
     char *outf = argv[2];
     char *echecksum = argv[3];
-
     char **args = calloc(argc-2, sizeof(void*));
 
     args[0] = cmd;
     for(int i=1; i<argc-3; i++)
         args[i] = untaint(argv[i+3]);
 
     int status;
+#ifdef _WIN32
+    char* fullcmd = malloc(2000);
+    strcpy(fullcmd, args[0]);
```

Comment:
Can we use a safe version of this, please? We're hardcoding a malloc of 2000, and args[0] might be longer than 2000 causing us to write beyond the end of the memory we've allocated.